### PR TITLE
[feature#14818] alert-spi support DingTalkPersonalWork

### PR DESCRIPTION
**Pull request by xjlgod**
## Purpose of the pull request

to add dinding talk personal work notification #14818

Implementation Mechanism：
[enterprise-created-chatbot](https://open.dingtalk.com/document/orgapp/enterprise-created-chatbot)
[chat-messages](https://open.dingtalk.com/document/orgapp/chatbot-sends-queries-and-withdraws-one-on-one-chat-messages)

## Brief change log

a new moudle dolphinscheduler-alert-dingtalk-personalwork be added

## Verify this pull request

create alert instance to test it

7126a57c3d5116106c3bce6661667e7

### CODE REVIEW

**Comment by melin** [comment]
support feishu?

**Comment by xjlgod** [comment]
> support feishu?

This pr not support feishu，I will do it in another pr.

**Review by songjianet — DISMISSED** [review]

**Comment by xjlgod** [comment]
I see SonarCloud check and OWASP did not pass. Should I add more unit test? I think OWASP is not about my pr.

**Comment by xjlgod** [comment]
Should I put the code into dolphinscheduler-alert-dingtalk moudle？

File: dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-dingtalk-personalwork/src/main/java/org/apache/dolphinscheduler/plugin/alert/dingtalk/personalwork/DingTalkPersonalWorkSender.java [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * <p>
+ *     https://open.dingtalk.com/document/orgapp/enterprise-created-chatbot
+ *     https://open.dingtalk.com/document/orgapp/chatbot-sends-queries-and-withdraws-one-on-one-chat-messages
+ * </p>
+ */
+@Slf4j
+public class DingTalkPersonalWorkSender {
```
Comment (SbloodyS) [inline dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-dingtalk-personalwork/src/main/java/org/apache/dolphinscheduler/plugin/alert/dingtalk/personalwork/DingTalkPersonalWorkSender.java:49]: Same as https://github.com/apache/dolphinscheduler/pull/14982#discussion_r1361751343

Comment (xjlgod) [reply dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-dingtalk-personalwork/src/main/java/org/apache/dolphinscheduler/plugin/alert/dingtalk/personalwork/DingTalkPersonalWorkSender.java:49]: > Same as [#14982 (comment)](https://github.com/apache/dolphinscheduler/pull/14982#discussion_r1361751343)

If we do this, their parameters will appear confusing when placed in a form. Is there a better way?  I think the best user experience is that after modifying `sendtype`, the form parameters will also be modified.

image
image

Comment (xjlgod) [reply dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-dingtalk-personalwork/src/main/java/org/apache/dolphinscheduler/plugin/alert/dingtalk/personalwork/DingTalkPersonalWorkSender.java:49]: > > Same as [#14982 (comment)](https://github.com/apache/dolphinscheduler/pull/14982#discussion_r1361751343)
> 
> If we do this, their parameters will appear confusing when placed in a form. Is there a better way? I think the best user experience is that after modifying `sendtype`, the form parameters will also be modified.
> 
> image image

@SbloodyS please check this and #https://github.com/apache/dolphinscheduler/pull/14982#issuecomment-1777144300

**Pull request closed without merging**